src/v/cluster/members_manager.cc

```diff
@@ -254,7 +254,9 @@ ss::future<> members_manager::handle_raft0_cfg_update(
         _in_progress_updates.erase(id);
     }
 
-    if (update_offset >= _last_connection_update_offset) {
+    if (
+      update_offset >= _last_connection_update_offset
+      && !fully_removed_nodes.empty()) {
```

Comment:
I wonder why this is needed. In theory it shouldn't matter if we update `_last_connection_update_offset` here even with `fully_removed_nodes` empty.

Reply:
@mmaslankaprv can you elaborate on this? Is this just an optimization?